cylc/flow/terminal.py

```diff
@@ -126,7 +126,7 @@ def ansi_log(name='cylc', stream='stderr'):
     stream_name = f'<{stream}>'
     for handler in logging.getLogger(name).handlers:
         if (
-            getattr(handler, 'formatter')
+            getattr(handler, 'formatter', None)
```

Comment:
Genuine bug, must specify a default for `getattr` to swallow the `AttributeError`:

```python
>>> getattr(None, 'x')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'x'
```

Reply:
Could use `hasattr`?

Reply:
Could indeed.